src/struct/EmoteFetcher.js

```diff
@@ -318,6 +318,36 @@ class EmoteFetcher {
             return this.channels.get(channel).emotes.filter(e => e.type === '7tv');
         });
     }
+
+    /**
+     * Converts emote JSONs to emotes
+     * @param {Object} [json] - Emote JSON
+     * @returns {Emote[]}
+     */
+    fromJSON(json) {
+        const emotes = [];
+        const classMap = {
+            bttv: { class: BTTVEmote, cache: channel_id => this._cacheBTTVEmote(channel_id) },
+            ffz: { class: FFZEmote, cache: channel_id => this._cacheFFZEmote(channel_id) },
+            '7tv': { class: SevenTVEmote, cache: channel_id => this._cacheSevenTVEmote(channel_id) },
+            twitch: { class: TwitchEmote, cache: channel_id => this._cacheTwitchEmote(channel_id) }
+        };
+        for (const emoteJSON of json) {
+            const { type } = emoteJSON;
+            const emoteClass = classMap[type].class;
```

Comment:
Just curious as when I used the `?` ESLint complained, it should be supported with node v16 no? Wondering if something has to be updated to allow for it

Reply:
Ah sorry, I typed that out during a break at work without checking with ESLint. Optional chaining is available since Node 14.
`.eslint.json` need to be updated with `"ecmaVersion": 2020` instead of 2018.